wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -1354,4 +1355,53 @@ public function autofillReceiptFrom($formName, &$form) {
     }
   }
 
+  /**
+   *
+   */
+  public static function updateCampaignForCollectionSourceContribution(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($objectName != 'Contribution' || $objectId === NULL) {
+      return;
+    }
+
+    $contributionId = $objectRef['id'] ?? NULL;
+    if (!$contributionId) {
+      return;
+    }
+
+    $contribution = Contribution::get(FALSE)
```

Comment:
How are we getting the contribution? Are we running the code on create or edit?

Reply:
I logged some data there where I was able to find the contribution id with the help of that I was able to find details